ipaclient/plugins/vault.py

```diff
@@ -169,6 +180,58 @@ def decrypt(data, symmetric_key=None, private_key=None):
                 message=_('Invalid credentials'))
 
 
+def get_cached_transport_cert(domain):
+    basename = DNSName(domain).ToASCII() + '.pem'
+    filename = os.path.join(TRANSPORT_CERT_CACHE_PATH, basename)
+
+    try:
+        try:
+            return x509.load_certificate_from_file(filename)
+        except EnvironmentError as e:
+            if e.errno != errno.ENOENT:
+                raise
+    except Exception as e:
+        logger.warning("Failed to load %s: %s", filename, e)
```

Comment:
Please use ```exc_info=True``` everywhere in order to get a full traceback. It makes debugging much more pleasant.